cschlaffner/PROTzilla2
# 464 unify uniprot protein graph data across runs

## Description
refere to #464 

## Changes
protein_graphs.py: changes to functionality -> these are the relevant changes. please specifically pay attention to directory creation and the control flow i.e. what is ensured/checked where


## Testing

1. create a protein graph in run A for e.g. Protein P10636 (example data set BA46 works fine)
2. create the same protein graph for P10636 in a new, separate run B
3. check the logs, file explorer, whether the files for P10636 have been downloaded again and whether the unmodified graph generated in run A has been used in run B to match peptides


## PR checklist
**Development**
 
**Mergeability**
- [x] The code has been formatted with `black`
 
**Code review**
